### PR TITLE
check empty table name in client

### PR DESCRIPTION
**Pull request by foronedream**
check empty table name
## Summary

check empty table name

## Solution Description

check empty table name in client

### CODE REVIEW

**Comment by foronedream** [comment]
需要检查是否所有的case都覆盖到了

File: src/test/java/com/alipay/oceanbase/rpc/ObTableClientTest.java [hunk 1]
```diff
@@ -2489,4 +2489,54 @@ PRIMARY KEY (`c1`)
...
+        /*
+         CREATE TABLE `test_query_filter_mutate` (`c1` bigint NOT NULL, `c2` varbinary(1024) DEFAULT NULL,
+                                                  `c3` varchar(20) DEFAULT NULL,`c4` bigint DEFAULT NULL, PRIMARY KEY(`c1`))
+                                                  partition by range columns (`c1`) ( PARTITION p0 VALUES LESS THAN (300),
+                                                  PARTITION p1 VALUES LESS THAN (1000), PARTITION p2 VALUES LESS THAN MAXVALUE);
+         */
+        final String TABLE_NAME = "test_query_filter_mutate";
+        System.setProperty("ob_table_min_rslist_refresh_interval_millis", "1");
+
+        ((ObTableClient) client).addRowKeyElement(TABLE_NAME, new String[] { "c1" });
```
Comment (shenyunlong) [inline src/test/java/com/alipay/oceanbase/rpc/ObTableClientTest.java:2505]: why set tableName equals to test_query_filter_mutate and addRowKeyElement for table test_query_filter_mutate, it  seems a test just for empty table?

Comment (foronedream) [reply src/test/java/com/alipay/oceanbase/rpc/ObTableClientTest.java]: already edited

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java [hunk 2]
```diff
@@ -928,6 +928,9 @@ public void syncRefreshMetadata() throws Exception {
      */
     public String getIndexTableName(final String dataTableName, final String indexName,
```
[the remark this answers is not in the document]
Comment (foronedream) [reply src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java]: already edited

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java [hunk 3]
```diff
@@ -1206,6 +1224,9 @@ private TableEntry refreshTableEntry(TableEntry tableEntry, String tableName)
      */
     private TableEntry refreshTableEntry(TableEntry tableEntry, String tableName, boolean fetchAll)
                                                                                                    throws ObTableEntryRefreshException {
+        if (tableName == null || tableName.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (foronedream) [reply src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java]: already edited

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java [hunk 4]
```diff
@@ -518,6 +518,9 @@ private <T> T execute(String tableName, TableExecuteCallback<T> callback) throws
      */
     private <T> T execute(String tableName, TableExecuteCallback<T> callback, ObServerRoute route)
                                                                                                   throws Exception {
+        if (tableName == null || tableName.isEmpty()) {
+            throw new IllegalArgumentException("table client with empty table name");
```
[the remark this answers is not in the document]
Comment (foronedream) [reply src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java]: already edited

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java [hunk 5]
```diff
@@ -1401,6 +1422,9 @@ public ObPair<Long, ObTableParam> getTable(String tableName, Object[] rowKey, bo
     public ObPair<Long, ObTableParam> getTable(String tableName, Object[] rowKey, boolean refresh,
                                                boolean waitForRefresh, ObServerRoute route)
                                                                                            throws Exception {
+        if (tableName == null || tableName.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (foronedream) [reply src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java]: already edited

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java [hunk 6]
```diff
@@ -1489,6 +1516,9 @@ public ObPair<Long, ObTableParam> getTable(String tableName, long partId, boolea
     public ObPair<Long, ObTableParam> getTable(String tableName, TableEntry tableEntry,
                                                long partId, boolean waitForRefresh,
                                                ObServerRoute route) throws Exception {
+        if (tableName == null || tableName.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (foronedream) [reply src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java]: already edited

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java [hunk 7]
```diff
@@ -1617,7 +1647,10 @@ public List<ObPair<Long, ObTableParam>> getTables(String tableName, Object[] sta
                                                       boolean endInclusive, boolean refresh,
                                                       boolean waitForRefresh, ObServerRoute route)
                                                                                                   throws Exception {
-
+        // check tablename value
+        if (tableName == null || tableName.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (foronedream) [reply src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java]: already edited

File: src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java [hunk 8]
```diff
@@ -1674,6 +1707,9 @@ public List<ObPair<Long, ObTableParam>> getTables(String tableName, Object[] sta
      * @return ObTableAggregation object
      */
     public ObTableAggregation aggregate(String tableName) {
+        if (tableName == null || tableName.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (foronedream) [reply src/main/java/com/alipay/oceanbase/rpc/ObTableClient.java]: already edited

**Comment by WeiXinChan** [comment]
LGTM

**Pull request merged**